src/middlewared/middlewared/plugins/disk_/wipe.py

```diff
@@ -16,25 +16,45 @@
 class DiskService(Service):
 
     @private
-    def get_partitions_quick(self, dev_name):
+    def get_partitions_quick(self, dev_name, tries=None):
         """
         Lightweight function to generate a dictionary of
-        partition start in units of bytes to be used by seek
+        partition start in units of bytes.
+
+        `tries` int, specifies the number of tries that we will
+            look for the various files in sysfs. Often times this
+            function is called after a drive has been formatted
+            and so the caller might want to wait on udev to become
+            aware of the new partitions.
         """
+        if tries in (0, 1) or not isinstance(tries, int):
+            tries = 1
+        else:
+            tries = min(tries, 10)
+
         startsect = {}
         sectsize = 0
-        lbs_file = f"/sys/block/{dev_name}/queue/logical_block_size"
-        try:
-            with open(lbs_file, "r") as fd:
-                sectsize = int(fd.read().strip())
-        except (FileNotFoundError, ValueError):
-            pass
-        except Exception:
-            self.logger.error('Unexpected failure trying to open %s', lbs_file, exc_info=True)
-        else:
-            for sdpath in glob(f"/sys/block/{dev_name}/{dev_name}[1-9]"):
-                with open(f"{sdpath}/start", "r") as fd:
-                    startsect[int(sdpath[-1])] = int(fd.read().strip()) * sectsize
+        path_obj = pathlib.path(f"/sys/block/{dev_name}")
+        for _try in range(tries):
+            if startsect:
+                # dictionary of partition info has already been populated
+                # so we'll break out early
+                return startsect
+            else:
+                time.sleep(0.5)
+
+            try:
+                sectsize = int((path_obj / 'queue/logical_block_size').read_text().strip())
+                with os.scandir(path_obj) as dir_contents:
+                    for partdir in filter(lambda x: x.is_dir() and x.name.startswith(dev_name), dir_contents):
+                        part_num = int((partdir / 'partition').read_text().strip())
+                        part_start = int((partdir / 'start').read_text().strip()) * sectsize
+                        startsect[part_num] = part_start
+            except (FileNotFoundError, ValueError):
+                continue
+            except Exception:
+                if _try == tries:
```

Comment:
Shouldn't this be
`if _try == tries - 1:`
to catch the last time round the loop?

Reply:
`range()` being half-open is always fun :smile: (It should be `tries + 1`)

Reply:
You say "potato".

Reply:
Fixed it nonetheless, thanks!